cpp/include/gar/graph.h

```diff
@@ -656,32 +645,32 @@ class EdgesCollection<AdjListType::ordered_by_source> {
                               fs->GetFileNumOfDir(base_dir));
     std::vector<IdType> edge_chunk_nums(vertex_chunk_num, 0);
     chunk_end_ = 0;
+    edge_num_ = 0;
     for (IdType i = 0; i < vertex_chunk_num; ++i) {
       GAR_ASSIGN_OR_RAISE_ERROR(
           edge_chunk_nums[i],
           utils::GetEdgeChunkNum(prefix, edge_info, adj_list_type_, i));
       chunk_end_ += edge_chunk_nums[i];
+      GAR_ASSIGN_OR_RAISE_ERROR(
+          auto chunk_edge_num_,
+          utils::GetEdgeNum(prefix, edge_info, adj_list_type_, i));
+      edge_num_ += chunk_edge_num_;
     }
     index_converter_ =
         std::make_shared<util::IndexConverter>(std::move(edge_chunk_nums));
-    offset_of_chunk_begin_ = 0;
-    offset_of_chunk_end_ = 0;
   }
 
   /**
    * @brief Initialize the EdgesCollection with a range of chunks.
    *
    * @param edge_info The edge info that describes the edge type.
    * @param prefix The absolute prefix.
-   * @param chunk_begin The global index of the begin chunk.
-   * @param chunk_end The global index of the end chunk.
+   * @param vertex_chunk_begin The index of the begin vertex chunk.
+   * @param vertex_chunk_end The index of the end vertex chunk (not included).
    */
   EdgesCollection(const EdgeInfo& edge_info, const std::string& prefix,
-                  IdType chunk_begin, IdType chunk_end)
-      : edge_info_(edge_info),
-        prefix_(prefix),
-        chunk_begin_(chunk_begin),
-        chunk_end_(chunk_end) {
+                  IdType vertex_chunk_begin, IdType vertex_chunk_end)
```

Comment:
Can we unify the three constructors into one and use default parameter like (vertex_chunk_begin = 0, vertex_chunk_end=std::numeric_limits<int64_t>::max()), then we can do some checks to match different cases:
if vertex_chunk_end == std::numeric_limits<int64_t>::max(): means the end is the last edge + 1 of last vertex chunk;

Reply:
In the original constructors, there are not checking that begin and end are valid. maybe we need to add some check to ensure:
- begin <= vertex_chunk_end
- vertex_chunk_begin <= vertex_chunk_num && vertex_chunk_end <= vertex_chunk_num

Reply:
> Can we unify the three constructors into one and use default parameter like (vertex_chunk_begin = 0, vertex_chunk_end=std::numeric_limits<int64_t>::max()), then we can do some checks to match different cases: if vertex_chunk_end == std::numeric_limits<int64_t>::max(): means the end is the last edge + 1 of last vertex chunk;

Good suggestion! I have unified the constructors.

Reply:
> In the original constructors, there are not checking that begin and end are valid or not. maybe we need to add some check to ensure:
> 
> * begin <= vertex_chunk_end
> * vertex_chunk_begin <= vertex_chunk_num && vertex_chunk_end <= vertex_chunk_num

Perhaps we do not need to add checks here, for invalid input parameters for vertex_chunk_begin and vertex_chunk_end, an empty EdgesCollection will be returned. Related tests have been added.